vllm/v1/core/kv_cache_manager.py

```diff
@@ -280,51 +269,12 @@ def _get_new_blocks(
         while idx < num_blocks:
             curr_block = self.free_block_queue.popleft()
             assert curr_block.ref_cnt == 0
-
-            # Evict blocks from the cache.
-            if self.enable_caching:
-                block_hash = curr_block.block_hash
-                if (block_hash is not None
-                        and block_hash in self.cached_block_hash_to_block):
-                    if len(self.cached_block_hash_to_block[block_hash]) == 1:
-                        del self.cached_block_hash_to_block[block_hash]
-                    else:
-                        del self.cached_block_hash_to_block[block_hash][
-                            curr_block.block_id]
-                curr_block.reset()
-
-            curr_block.ref_cnt = 1
+            curr_block.incr_ref()
             ret.append(curr_block)
             idx += 1
 
-        # Then assign token IDs to the allocated blocks.
```

Comment:
I removed many of the cache related outside of the `get_new_blocks` routine to make it lightweight.